File: java/link-message-client/src/main/java/link/message/client/content/complex/ComplexMessageContentItem.java
package link.message.client.content.complex;

import link.message.client.StringOrObjectSerializable;
import link.message.client.content.StringContent;
import link.message.client.utils.Guard;

import com.alibaba.fastjson.annotation.JSONField;

/**
 * 复杂消息内容的组成项
 * 
 * @author zhongt
 *
 */
public class ComplexMessageContentItem {
	protected StringOrObjectSerializable content;
	private String                     img;
	@JSONField(name="action_type")
	private int                        actionType = ActionType.NONE;
	@JSONField(name="action_param")
	private String                     actionParam;

	public ComplexMessageContentItem() {
		
	}
	
	public ComplexMessageContentItem(StringOrObjectSerializable content) {
		Guard.guardReqiredObject(content, "content must be set value.");
		
		this.content = content;
	}
	
	public ComplexMessageContentItem(String content) {
		Guard.guardReqiredString(content, "content must be set value.");
		
		this.content = new StringContent(content);
	}
	
	public ComplexMessageContentItem(String content, Action action) {
		this(content);
		
		Guard.guardReqiredObject(action,  "action object must be set value.");
		
		this.actionType  = action.getActionType();
		this.actionParam = action.getActionParam();
	}
	
	public ComplexMessageContentItem(StringOrObjectSerializable content, Action action) {
		this(content);
		
		Guard.guardReqiredObject(action, "action object must be set value.");
		
		this.actionType  = action.getActionType();
		this.actionParam = action.getActionParam();
	}
	
	public ComplexMessageContentItem(StringOrObjectSerializable content, String image, Action action) {
		this(content, action);
		
		this.img = image;
	}
	
	public ComplexMessageContentItem(String content, String image, Action action) {
		this(new StringContent(content), action);
		
		this.img = image;
	}
	
	public StringOrObjectSerializable getContent() {
		return content;
	}

	public void setContent(StringOrObjectSerializable content) {
		this.content = content;
	}

	public String getImg() {
		return img;
	}

	public void setImg(String img) {
		this.img = img;
	}

	public int getActionType() {
		return actionType;
	}

	public void setActionType(int actionType) {
		this.actionType = actionType;
	}

	public String getActionParam() {
		return actionParam;
	}

	public void setActionParam(String actionParam) {
		this.actionParam = actionParam;
	}
}

File: net/Link.Message.Client/content/complex/ComplexMessageContentItem.cs
using Link.Message.Client.utils;
using Newtonsoft.Json;

namespace Link.Message.Client.content.complex
{
	/// <summary>
	/// 复杂消息内容的组成项
	/// </summary>
	public class ComplexMessageContentItem
	{
		public IStringOrObjectSerializable Content { get; set; }
        [JsonProperty("img")]
        public string Img { get; set; }
        [JsonProperty("action_type")]
	    public ActionType ActionType { get; set; }
        [JsonProperty("action_param")]
	    public string ActionParam { get; set; }

		public ComplexMessageContentItem()
		{
		    ActionType = ActionType.None;
		}

		public ComplexMessageContentItem(IStringOrObjectSerializable content)
		{
			Guard.GuardReqiredObject(content, "content must be set value.");

			Content = content;
		}

		public ComplexMessageContentItem(string content)
		{
			Guard.GuardReqiredString(content, "content must be set value.");

			Content = new StringContent(content);
		}

		public ComplexMessageContentItem(string content, Action action) : this(content)
		{
			Guard.GuardReqiredObject(action, "action object must be set value.");

			ActionType = action.ActionType;
			ActionParam = action.ActionParam;
		}

		public ComplexMessageContentItem(IStringOrObjectSerializable content, Action action) : this(content)
		{

			Guard.GuardReqiredObject(action, "action object must be set value.");

			ActionType = action.ActionType;
			ActionParam = action.ActionParam;
		}

		public ComplexMessageContentItem(IStringOrObjectSerializable content, string image, Action action) : this(content, action)
		{
            Img = image;
		}

		public ComplexMessageContentItem(string content, string image, Action action) : this(new StringContent(content), action)
		{
			Img = image;
		}
	}
}